components/camera/server.go

```diff
@@ -75,45 +69,17 @@ func (s *serviceServer) GetImage(
 			req.MimeType = utils.MimeTypeJPEG
 		}
 	}
-
 	req.MimeType = utils.WithLazyMIMEType(req.MimeType)
 
-	ext := req.Extra.AsMap()
-	ctx = NewContext(ctx, ext)
-
-	var img image.Image
-	var release func()
-	switch castedCam := cam.(type) {
-	case ReadImager:
-		// RSDK-8663: If available, call a method that reads exactly one image. The default
-		// `ReadImage` implementation will otherwise create a gostream `Stream`, call `Next` and
-		// `Close` the stream. However, between `Next` and `Close`, the stream may have pulled a
-		// second image from the underlying camera. This is particularly noticeable on camera
-		// clients. Where a second `GetImage` request can be processed/returned over the
-		// network. Just to be discarded.
-		img, release, err = castedCam.Read(ctx)
-	default:
-		img, release, err = ReadImage(gostream.WithMIMETypeHint(ctx, req.MimeType), cam)
-	}
+	resBytes, resMetadata, err := cam.Image(ctx, req.MimeType, req.Extra.AsMap())
 	if err != nil {
 		return nil, err
 	}
-	defer func() {
-		if release != nil {
-			release()
-		}
-	}()
-
-	actualMIME, _ := utils.CheckLazyMIMEType(req.MimeType)
-	resp := pb.GetImageResponse{
-		MimeType: actualMIME,
-	}
-	outBytes, err := rimage.EncodeImage(ctx, img, req.MimeType)
-	if err != nil {
-		return nil, err
+	if len(resBytes) == 0 {
+		return nil, fmt.Errorf("received empty bytes from Image method of %s", req.Name)
 	}
-	resp.Image = outBytes
-	return &resp, nil
+	actualMIME, _ := utils.CheckLazyMIMEType(resMetadata.MimeType)
```

Comment:
Note: This will change behavior. 

Lets say you have a viam-server and camera hosted in a module.

Lets say that the camera in the module can only product PNGs but the user ask for a jpeg. 

Previously, the server.go in the module would convert the PNG into a JPEG before sending it to viam-server. 

Now, with this change, it will return the PNG to the caller.

I think this is still the right change, as implicitly converting between image formats can introduce image quality degridation as well as significantly worse performance (like 15X worse performance), but I want to confirm:

@bhaney & @randhid are we confident no customer robots depend on this `implicitly convert between image mime types` behavior?

Reply:
The customers I am aware of rely on video storage and upload and rtsp passthrough where they don't really ask for a specific mime type anyway.

Still trying to track down where gostream actually sets a default for both stream types we use.